### DISCUSSION TITLE
It's not clear how to use shiv without dependencies

### DISCUSSION THREAD

**Discussion by vitoyucepi** [linkedin/shiv#232] [Q&A]
## Summary
By default, shiv implies that I should supply the project with `setup.py` or `pyproject.toml` or `--site-packages` arg.

## To reproduce
1. `python -m venv .venv`
1. `. .venv/bin/activate`
1. `pip install shiv`
1. `echo "print('hello world')" >> __main__.py`
1. `shiv -e __main__.py -o main`

## Question
How can I use shiv without dependencies and project files?

**Comment by lorencarvalho** [comment]
hi @vitoyucepi,

I'm not sure if there is a compelling reason _to_ use `shiv` if you have no desire or need to ship dependencies. Can you share more about your use case & what you're hoping to achieve?

I would assume sans dependencies, one could just use Python's native `zipapp` functionality. https://docs.python.org/3/library/zipapp.html

**Reply by vitoyucepi** [reply]
Hi @lorencarvalho,
The reason for using `shiv` like I described it is to create a reproduction example for #231.
